File: apzkr-pzpi-21-2-butsulin-ihor/Task4-MobileClient/APZMobile/app/src/main/java/com/osigram/apz2024/mobile/models/Slot.kt
package com.osigram.apz2024.mobile.models

import kotlinx.serialization.Serializable

@Serializable
data class Slot(val ID: ULong, val maxWeight: Double, val item: Item?, val itemID: ULong?, val device: Device?, val weighingResults: List<WeighingResult>?, val storageID: ULong)

File: apzkr-pzpi-21-2-butsulin-ihor/Task1-Server/apz-backend/types/models/Slot.go
package models

import "gorm.io/gorm"

type Slot struct {
	gorm.Model
	MaxWeight       float64          `json:"maxWeight"`
	Item            *Item            `json:"item"`
	ItemID          *uint            `json:"itemID"`
	Device          *Device          `json:"device"`
	WeighingResults []WeighingResult `json:"weighingResults"`
	StorageID       uint             `json:"storageID"`
}
